Dockerfile

```
@@ -1,35 +1,56 @@
-# build stage
-FROM golang:1-alpine AS build-env
-RUN apk add --no-cache \
-    build-base \
-    linux-headers \
-    coreutils \
-    binutils \
-    libtool \
-    musl-dev \
-    git \
-    make \
-    gcc \
-    libc-dev \
-    zip \
-    ca-certificates \
-    libcap
-
-ENV GO111MODULE=on \
-    CGO_ENABLED=0
-    
-WORKDIR /src
+# get newest certificates
+FROM --platform=$BUILDPLATFORM alpine:3.16 AS ca-certs
+RUN apk add --no-cache ca-certificates
+RUN --mount=type=cache,target=/etc/ssl/certs \
+    update-ca-certificates 2>/dev/null || true
 
+# zig compiler
+FROM --platform=$BUILDPLATFORM ghcr.io/euantorano/zig:master AS zig-env
+
+# build environment
+FROM --platform=$BUILDPLATFORM golang:1-alpine AS build
+
+# required arguments(buildx will set target)
+ARG VERSION
+ARG BUILD_TIME
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETVARIANT
+
+# set working directory
+WORKDIR /go/src
+
+# download packages
 COPY go.mod go.sum ./
-RUN go mod download
+RUN --mount=type=cache,target=/go/pkg \
+    go mod download
 
 # add source
-ADD . .
+COPY . .
 
-ARG opts
-RUN env ${opts} make build-static && \
-    chown 100 /src/bin/blocky && \
-    setcap 'cap_net_bind_service=+ep' /src/bin/blocky
+# setup go+zig
+COPY --from=zig-env /usr/local/bin/zig /usr/local/bin/zig
+ENV PATH="/usr/local/bin/zig:${PATH}" \
+    CC="zigcc" \
+    CXX="zigcpp" \
+    CGO_ENABLED=0 \
+    GOOS="linux" \
+    GOARCH=$TARGETARCH \
+    GO_SKIP_GENERATE=1\
+    GO_BUILD_FLAGS="-tags static -v " \
+    BIN_USER=100\
+    BIN_AUTOCAB=1 \
+    BIN_OUT_DIR="/bin"
+
+#add make & libcap
+RUN apk add --no-cache make libcap
+
+# build binary 
+RUN --mount=type=bind,target=. \
+    --mount=type=cache,target=/root/.cache/go-build \ 
+    --mount=type=cache,target=/go/pkg \
+    export GOARM=${TARGETVARIANT##*v} && \
+    make build
```

Looking at the commits I think you already tried this, but I'm sure it's supposed to work:
```suggestion
    make build GOARM="${TARGETVARIANT##*v}"
```

Other possibility:
```suggestion
    GOARM="${TARGETVARIANT##*v}" make build
```